### PR TITLE
Run IWYU to clean up include and add platform-dependent files relative to main source directory

### PR DESCRIPTION
**Pull request by mayurdahibhate**
Fixed the issues listed on [#569](https://github.com/htop-dev/htop/issues/569).

Following changes suggested by ```IWYU``` were not made :
1. ```DiskIOMeter.h```, ```NetworkIOMeter.h``` and ```Panel.h``` produces warning after applying suggested changes.
2. ```linux/HugePageMeter.c``` produces error after applying the suggested changes.
3. ```linux/LinuxProcessList.c``` has a comment on the same line as the include suggesting to keep it.
4. Struct suggestions for ```Action.h``` and ```Process.h``` were ignored.

Added second blank line after all the headers if not present. Platform-dependent files were included relative to the main source directory with exceptions suggested by @BenBE .

### CODE REVIEW

File: dragonflybsd/Platform.c [hunk 1]
```diff
@@ -6,7 +6,16 @@ Released under the GNU GPLv2, see the COPYING file
...
 */
 
-#include "Platform.h"
+#include "dragonflybsd/Platform.h"
+
+#include <math.h>
+#include <time.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <sys/time.h>
```
Comment (BenBE) [inline dragonflybsd/Platform.c:15]: Is order important here? If so, please document with a comment.

File: solaris/SolarisProcessList.h [hunk 2]
```diff
@@ -13,8 +13,6 @@ in the source distribution for its full text.
 #define GZONE "global    "
 #define UZONE "unknown   "
```
Comment (BenBE) [inline solaris/SolarisProcessList.h:14]: Any reason those constants are above the includes? Documentation would be good for this.

**Review by BenBE — CHANGES_REQUESTED** [review]
Looks mostly good already. There's some compilation issues (cf. LGTM.com for details).

File: dragonflybsd/Platform.c (hunk 1 above)
Comment (mayurdahibhate) [reply dragonflybsd/Platform.c]: The order is not important here. The order is fixed in the latest commit according to ```STYLEGUIDE```.

File: solaris/SolarisProcessList.h (hunk 2 above)
Comment (mayurdahibhate) [reply solaris/SolarisProcessList.h:14]: I am unsure why those constants are above the includes. But I think defining them after includes won't break anything as ```GZONE``` and ```UZONE``` are passed to ```xstrdup()``` .

File: dragonflybsd/Platform.c [hunk 3]
```diff
@@ -10,8 +10,8 @@ in the source distribution for its full text.
 
 #include <math.h>
 #include <time.h>
-#include <sys/types.h>
 #include <sys/sysctl.h>
+#include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
```
Comment (BenBE) [inline dragonflybsd/Platform.c:16]: Order?

File: linux/Platform.c [hunk 4]
```diff
@@ -66,6 +66,7 @@ in the source distribution for its full text.
 
 #ifdef HAVE_SENSORS_SENSORS_H
 #include "LibSensors.h"
+#include <errno.h>
```
Comment (BenBE) [inline linux/Platform.c:69]: Order? Is it only used with `HAVE_SENSORS_SENSORS_H` active?

File: dragonflybsd/Platform.c (hunk 3 above)
Comment (mayurdahibhate) [reply dragonflybsd/Platform.c]: Fixed this time.

File: linux/Platform.c (hunk 4 above)
Comment (mayurdahibhate) [reply linux/Platform.c]: After seeing [lgtm](https://lgtm.com/projects/g/htop-dev/htop/logs/rev/pr-14d11c36ce175c408cc1ff7a16a95e0c50a49c85/lang:cpp/stage:Build%20merge_628a5fa050b2d91cdd8989404e4aa1f57f23e852) log. I thought I was supposed to use ```errno.h``` when ```HAVE_SENSORS_SENSORS_H``` is active. On furthur inspection I found that it is supposed to be active when ```HAVE_LIBCAP``` is active.

**Comment by BenBE** [comment]
> Hello, @BenBE. I am not sure what I am supposed to do next. I am new to the open source and to this project. Any hint/help would be appreciated.

Sorry for the late reply as I've been busy with some other PR, thus I haven't come around to look at this again yet. That other PR (#388) is also blocking all larger code changes right now. Depending on how well I can rebase the changes there I might prioritize merging this one, but I have to check.

From your side there's nothing to be done ATM. I'll get back if something comes up.

**Review by BenBE — APPROVED** [review]

**Pull request merged**